gen/nvml/refcount.go

```diff
@@ -0,0 +1,43 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package nvml
+
+import ()
+
+type refcount int
+
+func (r *refcount) Inc() {
+	(*r)++
+}
+
+func (r *refcount) Dec() {
+	if *r > 0 {
+		(*r)--
+	}
+}
```

Comment:
What about:
```suggestion
func (r *refcount) Dec() func(error) {
	if *r > 0 {
		(*r)--
		return r.IncOnError
	}
	// return a no-op if we don't decrease refcount.
	return func(error) {}
}
```

And then we would use this:
```
defer l.refcount.Dec()(rerr)
```
?

Reply:
This is less readable to me than just having two separate calls.

Reply:
I meant that this handles the zero case, but sure.